packages/dd-trace/src/llmobs/tagger.js

```diff
@@ -78,32 +78,29 @@ class LLMObsTagger {
   }
 
   tagMetadata (span, metadata) {
-    try {
-      span.setTag(METADATA, JSON.stringify(metadata))
-    } catch {
-      logger.warn('Failed to parse span metadata. Metadata key-value pairs must be JSON serializable.')
-    }
+    span.setTag(METADATA, metadata)
   }
 
   tagMetrics (span, metrics) {
-    try {
-      span.setTag(METRICS, JSON.stringify(metrics))
-    } catch {
-      logger.warn('Failed to parse span metrics. Metrics key-value pairs must be JSON serializable.')
+    const filterdMetrics = {}
+    for (const [key, value] of Object.entries(metrics)) {
+      if (typeof value === 'number') {
+        filterdMetrics[key] = value
+      } else {
+        logger.warn(`Metric ${key} must be a number, instead got ${value}`)
```

Comment:
nit; maybe put quotation marks around `${key}` / rephrase it like
"Value for metric '{metric_name}' must be a number, instead got .."